enhancements/scheduling/priority-class.md

```diff
@@ -0,0 +1,110 @@
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in
+  [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+By default OpenShift has two reserved priority classes for critical system pods
+to have guaranteed scheduling, this proposal suggests the need for a third less
+critical priority class which would be used for user facing features we ship
+with OpenShift, for example user workload monitoring pods, that are not system
+critical but should have a higher scheduling priority than any user workloads.
+
+## Motivation
+
+Currently OpenShift monitoring ships with two workloads, user and cluster
+monitoring. These include prometheus pods, they both have the same priority
+classes set. What can and actually did happen, is that with not enough
+resources, user workload prometheus pods were scheduled in favour of the
+cluster monitoring prometheus pods. This should not happen, as cluster
+monitoring is system critical component, whereas user monitoring has less
+priority. 
+
+### Goals
+
+1. Describe the current state of pod priority classes in OpenShift.
+2. Introduce a new third priority class to be used within OpenShift by any non
+   critical workloads.
+
+### Non-Goals
+
+Out of scope of this proposal is solving user specific priority classes.
+
+## Proposal
+
+Priority indicates the importance of a Pod relative to other Pods. If a Pod
+cannot be scheduled, the scheduler tries to preempt (evict) lower priority Pods
+to make scheduling of the pending Pod possible.
+
+By default OpenShift has two reserved priority classes for critical system pods
+to have guaranteed scheduling: (following is from our docs [1]):
+- `system-cluster-critical` - This priority class has a value of 2000000000
+  (two billion) and is used with pods that are important for the cluster. Pods
+  with this priority class can be evicted from a node in certain circumstances.
+  For example, pods configured with the system-node-critical priority class can
+  take priority. However, this priority class does ensure guaranteed
+  scheduling.  Examples of pods that can have this priority class are fluentd,
+  add-on components like descheduler, and so forth.
+- `system-node-critical` - This priority class has a value of 2000001000 and is
+  used for all pods that should never be evicted from a node. Examples of pods
+  that have this priority class are sdn-ovs, sdn, and so forth.
+
+As mentioned before these two are not enough and this proposal suggests
+introducing a third priority class: `user-critical`. This would be used by any
+pods that are important for user facing OpenShift features but are not deemed
+system critical. Example of pods include user workload monitoring and user's
+Knative Service. 
```

Comment:
I'd replace "user's Knative Service" here with "the OpenShift Serverless control and data planes". We don't want OpenShift Serverless control or data planes to be system-*-critical because it's an optional component and not critical to the core functioning of the cluster itself. But, we do want the OpenShift Serverless control and data plane pods to be a higher priority than user workloads because if OpenShift Serverless control or data plane pods get evicted then that degrades the functionality of all Knative workloads in the cluster.